hosting/files/control-tower.js

```diff
@@ -0,0 +1,78 @@
+import init from "./base-init.js";
+import {
+    drawProductCardVonRequestURL,
+    drawProductCardVonLocalStorage,
+} from "./product-card-draw.js";
+import {
+    showSearchPage,
+    hideSearchPage,
+    showOnlySaleProduct,
+    hideOnlySaleProduct,
+    showMusinsaOnlyProduct,
+    showAllProductIncludeNotMusinsaOnly,
+    showSoldOutProduct,
+    hideSoldOutProduct,
+} from "./base-header-menu.js";
+import { addFilterTag, removeFilterTag } from "./tagging.js";
+
+export default function ControlTower() {
+    this.controlSwitches = {
+        searchOnOffSwitch: false,
+        saleProductSwitch: true,
+        musinsaOnlyProductSwitch: false,
+        includeSoldOutProductSwitch: false,
+        infiniteStone: true,
+    };
+}
+
+ControlTower.prototype.init = init(); // 첫 화면 구성.
+ControlTower.prototype.getDataForProducts = drawProductCardVonRequestURL(); // 최초 데이터를 받음.
+ControlTower.prototype.drawFirstCardList = drawProductCardVonLocalStorage(); // localStorage에 저장된 데이터를 카드형태로 화면에 뿌림.
+ControlTower.prototype.searchOnOff = function () {
+    if (this.controlSwitches.searchOnOffSwitch === false) {
+        showSearchPage();
+        return (this.controlSwitches.searchOnOffSwitch = true);
+    } else if (this.controlSwitches.searchOnOffSwitch === true) {
+        hideSearchPage();
+        drawProductCardVonLocalStorage();
+        return (this.controlSwitches.searchOnOffSwitch = false);
+    } else {
+        console.error();
+    }
+};
+ControlTower.prototype.saleProductOnOff = function () {
+    if (this.controlSwitches.saleProductSwitch === false) {
+        showOnlySaleProduct();
+        return (this.controlSwitches.saleProductSwitch = true);
+    } else if (this.controlSwitches.saleProductSwitch === true) {
+        addFilterTag("세일상품");
+        hideOnlySaleProduct();
+        return (this.controlSwitches.saleProductSwitch = false);

```

Comment:
위에서 리뷰 드린대로 saleProductSwitch 네이밍을 '할인 중인가?' 라는 의미의 `isOnSale`로 바꿨다고 생각해보면
이렇게 바꿔볼 수 있겠습니당

```suggestion
    if (this.controlSwitches.isOnSale) {
        addFilterTag("세일상품");
        hideOnlySaleProduct();
        this.controlSwitches.isOnSale = false
    } else if (!this.controlSwitches.isOnSale) {
        showOnlySaleProduct();
        this.controlSwitches.saleProductSwitch = true;
    }
```

if문 안에는 긍정문을 먼저 사용하는 게 이해하기 쉽더라고요!